compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MPQSOLVER_BISECTION_VISQ_ERROR_APPROXIMATOR_H__
+#define __MPQSOLVER_BISECTION_VISQ_ERROR_APPROXIMATOR_H__
+
+#include <string>
+#include <map>
+
+namespace mpqsolver
+{
+namespace bisection
+{
+
+class VISQErrorApproximator final
+{
+public:
+  /**
+   * @brief constructor of VISQErrorApproximator
+   */
+  VISQErrorApproximator() = default;
+
+  /**
+   * @brief initiliaze by visq_data_path (throws on failure)
+   */
+  void init(const std::string &visq_data_path);
+
+  /**
+   * @brief initiliaze by visq_data (returns success)
+   */
+  bool init(std::istream &visq_data);
```

Comment:
Is this method public for testing?

Reply:
yes. So it can be tested from `VISQErrorApproximator.test.cpp`

Reply:
I don't think adding a method inside `VISQErrorApproximator` just for testing is a good idea.

Reply:
`init` is moved to private section. tests will be compiled as `friend`